### DISCUSSION TITLE
Semantic segmentation on multi-dimensional Time Series with floss / fluss

### DISCUSSION THREAD

**Discussion by tknechtel** [TDAmeritrade/stumpy#540] [Help: Coding & Implementations]
Is there any possibility to segment multivariate dataset with floss (or fluss)- if yes, how?

For example: 
If i would like to segment the dataset provided in [Tutorial_Multidimensional_Motif_Discovery.html](https://stumpy.readthedocs.io/en/latest/Tutorial_Multidimensional_Motif_Discovery.html) with floss like in [Tutorial_Semantic_Segmentation.html](https://stumpy.readthedocs.io/en/latest/Tutorial_Semantic_Segmentation.html).

Any thanks for your help our tips in advance.

**Comment by seanlaw** [comment]
@tknechtel Thank you for your question. So, `floss`/`fluss` don't currently support the analysis of multidimensional matrix profiles Though, I haven't thought about it deeply enough but I have no idea how one would interpret the results from such an analaysis and what meaning it would have. Are you able to share more of what you are trying to accomplish and why you think semantic segmentation is the answer? Additionally, is there a reason why 1-D analysis wouldn't be sufficient?

**Reply by tknechtel** [reply]
I want to find the regimes in a multivariate data set. If I understand correctly, I should only select one matrix profile to do this. Or is it not possible to identify the regimes in a multivariate data set with the matrix profile?

**Reply by seanlaw** [reply]
@tknechtel Currently, there is no direct multi-dimensional support as `floss`/`fluss` only support 1-D inputs. Of course, you could attempt to compute the multi-dimensional matrix profile and then use the last row of your matrix profile indices as input to `floss`/`fluss`. Unfortunately, I have not tried this before and the interpretation of the results is beyond my experience and expertise so try at your own risk.

**Comment by JaKasb** [comment]
The input for FlOSS is the MPindex.
If you can compute the MPindex for a multivariate timeseries, then you can compute the parabola which is the core of FLOSS.
